src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvm/young_gen/YoungGenMissingMetricsTest.java

```diff
@@ -0,0 +1,352 @@
+
+/**
+ * Negative test: Tests that the action is NOT emitted if the required metrics are missing.
+ */
+
+@Category(RcaItMarker.class)
+@RunWith(RcaItNotEncryptedRunner.class)
+@AClusterType(ClusterType.MULTI_NODE_CO_LOCATED_MASTER)
+@ARcaGraph(ElasticSearchAnalysisGraph.class)
+@ARcaConf(dataNode = YoungGenITConstants.RCA_CONF_PATH, electedMaster =
+    YoungGenITConstants.RCA_MASTER_CONF_PATH)
+
+@AMetric(
+    name = Heap_Used.class,
+    dimensionNames = {Constants.TYPE_VALUE},
+    tables = {
+        @ATable(
+            hostTag = HostTag.DATA_0,
+            tuple = {
+                @ATuple(
+                    dimensionValues = {GCType.Constants.OLD_GEN_VALUE},
+                    sum = 950000000.0, avg = 950000000.0, min = 950000000.0, max = 950000000.0
+                )
+            }
+        ),
+        @ATable(
+            hostTag = HostTag.ELECTED_MASTER,
+            tuple = {
+                @ATuple(
+                    dimensionValues = {GCType.Constants.OLD_GEN_VALUE},
+                    sum = 950000000.0, avg = 950000000.0, min = 950000000.0, max = 950000000.0
+                )
+            }
+        )
+    }
+)
+@AMetric(
+    name = GC_Collection_Event.class,
+    dimensionNames = {Constants.TYPE_VALUE},
+    tables = {
+        @ATable(
+            hostTag = HostTag.DATA_0,
+            tuple = {
+                @ATuple(
+                    dimensionValues = {GCType.Constants.TOT_FULL_GC_VALUE},
+                    sum = 10.0, avg = 10.0, max = 10.0, min = 10.0
+                )
+            }
+        ),
+        @ATable(
+            hostTag = HostTag.ELECTED_MASTER,
+            tuple = {
+                @ATuple(
+                    dimensionValues = {GCType.Constants.TOT_FULL_GC_VALUE},
+                    sum = 10.0, avg = 10.0, max = 10.0, min = 10.0
+                )
+            }
+        )
+    }
+)
+@AMetric(
+    name = GC_Type.class,
+    dimensionNames = {GCInfoDimension.Constants.MEMORY_POOL_VALUE,
+        GCInfoDimension.Constants.COLLECTOR_NAME_VALUE},
+    tables = {
+        @ATable(
+            hostTag = HostTag.DATA_0,
+            tuple = {
+                @ATuple(
+                    dimensionValues = {GCType.Constants.OLD_GEN_VALUE,
+                        JvmSizingITConstants.CMS_COLLECTOR},
+                    sum = 10.0, avg = 10.0, max = 10.0, min = 10.0
+                )
+            }
+        ),
+        @ATable(
+            hostTag = HostTag.ELECTED_MASTER,
+            tuple = {
+                @ATuple(
+                    dimensionValues = {GCType.Constants.OLD_GEN_VALUE,
+                        JvmSizingITConstants.CMS_COLLECTOR},
+                    sum = 10.0, avg = 10.0, max = 10.0, min = 10.0
+                )
+            }
+        )
+    }
+)
+public class YoungGenMissingMetricsTest {
+
+  public static final int SLEEP_DURATION_IN_S = 190;
+
+  private TestApi testApi;
+
+  @Test
+  @AExpect(
+      what = Type.DB_QUERY,
```

Comment:
can we use rest endpoint to query action list instead ?

Reply:
Done.